nimpretty/nimpretty.nim

```diff
@@ -61,37 +62,61 @@ proc prettyPrint(infile, outfile: string, opt: PrettyOptions) =
     closeParsers(p)
 
 proc main =
-  var infile, outfile: string
+  var outfile, outdir: string
+
+  var infiles = newSeq[string]()
+  var outfiles = newSeq[string]()
+
   var backup = false
     # when `on`, create a backup file of input in case
     # `prettyPrint` could over-write it (note that the backup may happen even
     # if input is not actually over-written, when nimpretty is a noop).
     # --backup was un-documented (rely on git instead).
   var opt = PrettyOptions(indWidth: 0, maxLineLen: 80)
+
+
   for kind, key, val in getopt():
     case kind
     of cmdArgument:
-      infile = key.addFileExt(".nim")
+      infiles.add(key.addFileExt(".nim"))
     of cmdLongOption, cmdShortOption:
       case normalize(key)
       of "help", "h": writeHelp()
       of "version", "v": writeVersion()
       of "backup": backup = parseBool(val)
       of "output", "o", "out": outfile = val
+      of "outDir", "outdir", "d": outdir = val
       of "indent": opt.indWidth = parseInt(val)
       of "maxlinelen": opt.maxLineLen = parseInt(val)
       else: writeHelp()
     of cmdEnd: assert(false) # cannot happen
-  if infile.len == 0:
+  if infiles.len == 0:
     quit "[Error] no input file."
-  if outfile.len == 0:
-    outfile = infile
-  if not fileExists(outfile) or not sameFile(infile, outfile):
-    backup = false # no backup needed since won't be over-written
-  if backup:
-    let infileBackup = infile & ".backup" # works with .nim or .nims
-    echo "writing backup " & infile & " > " & infileBackup
-    os.copyFile(source = infile, dest = infileBackup)
-  prettyPrint(infile, outfile, opt)
+
+  if outfile.len != 0 and outdir.len != 0:
+    quit "[Error] out and outDir cannot both be specified"
+
+  if outfile.len == 0 and outdir.len == 0:
+    outfiles = infiles
+  elif outfile.len != 0 and infiles.len > 1:
+    # Take the last file to maintain backwards compatibility
+    let infile = infiles[^1]
+    infiles = @[infile]
+    outfiles = @[outfile]
+  elif outfile.len != 0:
+    outfiles = @[outfile]
+  elif outdir.len != 0:
+    outfiles = infiles.mapIt($(joinPath(outdir, it)))
+  
+  for (infile, outfile) in zip(infiles, outfiles):
+      let (dir, _, _) = splitFile(outfile)
```

Comment:
Indentation is 2 spaces.

Reply:
Ironic that I didn't run nimpretty on itself... It's fixed now.